### PR TITLE
Add pocket

### PR DESCRIPTION
**Pull request by gastonabella**
### Brief Description
Add pocket service for frontend.
### Related card or ticket
https://github.com/Pis-moove-it/reciclando-backend/issues/10
### Additional Details
Given a route_id, a collection_point_id and an array of serial_numbers, it creates a Collection associated to the ids mentioned, and create pockets with the serial numbers given associated to the collection too.
- Service for frontend
- Controller tests
- Swagger service
- Fixtures done for routes and collection points

### CODE REVIEW

File: spec/controllers/collections_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+RSpec.describe CollectionsController, type: :controller do
+  let(:json_response) { JSON.parse(response.body, symbolize_keys: true) }
+  let!(:organization) { FactoryBot.create(:organization) }
```
Comment (fbalarini) [inline spec/controllers/collections_controller_spec.rb]: Just a comment, you don't have to add FactoryBot as a prefix.

File: app/controllers/collections_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class CollectionsController < AuthenticateController
+  def create
+    collection = Collection.new(collection_params.merge(pockets_attributes: pockets_with_organization,
+                                                        route_id: params[:route_id]))
+    if collection.save
+      head :ok
```
Comment (fbalarini) [inline app/controllers/collections_controller.rb]: We should return the created collection

File: spec/controllers/collections_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                              collection: { collection_point_id: collection_point_id,
+                                            pockets_attributes: pocket_attributes } }
+    end
+    context 'when the user is authenticated' do
+      let!(:auth_user) { create_an_authenticated_user_with(organization, '1', 'android') }
+      let!(:auth_route) { FactoryBot.create(:route, user: auth_user) }
+      let(:invalid_route_id) { Route.pluck(:id).max + 1 }
+
+      context 'when creating valid collections' do
+        it 'does return success' do
```
Comment (fbalarini) [inline spec/controllers/collections_controller_spec.rb:23]: Add a test to check we are returning the created collection

File: app/controllers/collections_controller.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    else
+      render_error(1, collection.errors)
+    end
+  end
+
+  private
+
+  def pockets_with_organization
```
Comment (fbalarini) [inline app/controllers/collections_controller.rb:14]: It is ok for now but this should be changed to a callback in the near future.

File: spec/controllers/collections_controller_spec.rb (hunk 3 above)
Comment (gastonabella) [reply spec/controllers/collections_controller_spec.rb:23]: Should I return the pockets too? Or the collection is enough?

File: app/controllers/collections_controller.rb (hunk 4 above)
Comment (gastonabella) [reply app/controllers/collections_controller.rb:14]: I'll find out how to this with a callback after this PR was merged

**Review by accg14 — APPROVED** [review]

**Pull request merged**